root/relationship/linktype/RelationshipTypePairTree.js

```diff
@@ -0,0 +1,231 @@
+        {' ('}
+        <a className="toggle" style={{cursor: 'pointer'}}>
+          {l('more')}
+        </a>
+        {')'}
+      </span>
+
+      <div
+        className="reldetails"
+        style={{marginLeft: '20px', padding: '3px'}}
+      >
+        {$c.user?.is_relationship_editor ? (
+          <>
+            <strong>{l('Child order:')}</strong>
+            {' '}
+            {relType.child_order}
+            <br />
+          </>
+        ) : null}
+
+        <strong>{l('Forward link phrase:')}</strong>
+        {' '}
+        {relType.link_phrase
+          ? l_relationships(relType.link_phrase)
+          : lp('(none)', 'link phrase')}
+        <br />
+
+        <strong>{l('Reverse link phrase:')}</strong>
+        {' '}
+        {relType.reverse_link_phrase
+          ? l_relationships(relType.reverse_link_phrase)
+          : lp('(none)', 'link phrase')}
+        <br />
+
+        <strong>{l('Long link phrase:')}</strong>
+        {' '}
+        {relType.long_link_phrase
+          ? l_relationships(relType.long_link_phrase)
+          : lp('(none)', 'link phrase')}
+        <br />
+
+        <strong>{l('Description:')}</strong>
+        {' '}
+        {relType.description
+          ? l_relationships(relType.description)
+          : lp('(none)', 'description')}
+        <br />
+
+        <strong>{l('entity0 cardinality:')}</strong>
+        {' '}
+        <Cardinality cardinality={relType.cardinality0} />
+        <br />
+
+        <strong>{l('entity1 cardinality:')}</strong>
+        {' '}
+        <Cardinality cardinality={relType.cardinality1} />
+        <br />
+
+        <strong>{l('UUID:')}</strong>
+        {' '}
+        {relType.gid}
+        <br />
+
+        {'[ '}
+        {$c.user?.is_relationship_editor ? (
+          <>
+            <a href={'/relationship/' + relType.gid + '/edit'}>
+              {l('Edit')}
+            </a>
+            {childrenTypes.length ? null : (
+              <>
+                {' | '}
+                <a href={'/relationship/' + relType.gid + '/delete'}>
+                  {l('Remove')}
+                </a>
+              </>
+            )}
+            {' | '}
+          </>
+        ) : null}
+        {exp.l(
+          '{relationship_documentation_url|Documentation}',
+          {relationship_documentation_url: '/relationship/' + relType.gid},
+        )}
+        {' ]'}
+      </div>
+
+      {childrenTypes.length ? (
+        <ul>
+          {childrenTypes
+            .sort((a, b) => compareChildren(a, b))
+            .map(childrenType => (
+              <RelationshipTypeDetails
+                $c={$c}
+                key={childrenType.gid}
+                relType={childrenType}
+              />
+            ))}
+        </ul>
+      ) : null}
+    </li>
+  );
+};
+
+const RelationshipTypePairTree = ({
+  $c,
+  root,
+}: RelationshipTypePairTreeProps) => {
+  const childrenTypes = root.children || [];
+  const type0 = root.type0;
+  const type1 = root.type1;
```

Comment:
These were passed separately to the TT (and so was yet another argument, types) but I'm not sure why. Did I miss a reason why it's better?

Reply:
I can't see any reason for why the old way would be better (as long as this works).